e2e/workspace-core/src/run-commands.test.ts

```diff
@@ -143,4 +144,40 @@ describe('Run Commands', () => {
       })
     ).not.toThrow();
   }, 1000000);
+
+  it('should handle caching output directories containing trailing slashes', async () => {
+    const mylib = uniq('lib');
+
+    const folder = `dist/libs/${mylib}/some-folder`;
+
+    runCLI(`generate @nrwl/workspace:lib ${mylib}`);
+
+    runCLI(
+      `generate @nrwl/workspace:run-commands build --command=echo --outputs=${folder}/ --project=${mylib}`
+    );
+
+    const commands = [
+      process.platform === 'win32'
+        ? `mkdir ${folder}` // Windows
+        : `mkdir -p ${folder}`,
+      `echo dummy > ${folder}/dummy.txt`,
+    ];
+    updateProjectConfig(mylib, (config) => {
+      delete config.targets.build.options.command;
+      config.targets.build.options = {
+        ...config.targets.build.options,
+        parallel: false,
+        commands: commands,
+      };
+      return config;
+    });
+
+    // confirm that it builds correctly
+    runCLI(`build ${mylib}`);
```

Comment:
The lines 176-177 and 180-181 seem to be doing the same thing?!

What is it that you wanted to test here?

Reply:
We probably want to have assertions with and without a trailing slash in the same test to confirm both work equally.

Reply:
I will repeat the explanatory comment from implementation fix. The detail is in #10549 - the first run correctly builds whilst the second incorrectly populates from the cache (as a side effect of the trailing slash) resulting in an empty directory.

The issue is noted in the `copy()` method below:
```
// 'cp -a /path/dir/ dest/' operates differently to 'cp -a /path/dir dest/'
```

Regarding the assertions comment: The critical check is the `checkFilesExists()` which must be true after both the actual build (first run) and the populate from cache (second run). Without the fix, the second `checkFilesExists()` fails.

Reply:
Thank you, I was missing the context. Adding the comment will help future readers to understand why we run it twice.